mne/proj.py

```diff
@@ -97,15 +98,23 @@ def _compute_proj(data, info, n_grad, n_mag, n_eeg, desc_prefix, verbose=None):
         if n == 0:
             continue
         data_ind = data[ind][:, ind]
-        U = linalg.svd(data_ind, full_matrices=False,
-                       overwrite_a=True)[0][:, :n]
-        for k, u in enumerate(U.T):
+        # data is the covariance matrix: U * S**2 * Ut
+        U, Sexp2, _ = linalg.svd(data_ind, full_matrices=False,
+                                 overwrite_a=True)
+        U = U[:, :n]
+        exp_var_ = Sexp2 / n_epochs
```

Comment:
n_epochs is not needed as you divide by the sum below.

how about just logging the explained variance and not storing it as attribute?

note that the attribute will not be saved to disk.